subsys/gazell/Kconfig

```diff
@@ -109,23 +109,23 @@ choice GAZELL_TIMER
 
 config GAZELL_TIMER0
 	bool "TIMER0"
-	depends on !$(dt_nodelabel_enabled,timer0)
+	depends on $(dt_nodelabel_has_compat,timer0,$(DT_COMPAT_NORDIC_NRF_TIMER))
```

Comment:
Suggest checking 2 items as:
`depends on $(dt_nodelabel_has_compat,timer0,$(DT_COMPAT_NORDIC_NRF_TIMER))`
`depends on !$(dt_nodelabel_enabled,timer0)`

The first line checks that the SoC has the TIMER instance. Some nRF52 Series SoC do not have TIMER3 and TIMER4.
The second line checks that the Zephyr timer driver is not enabled.

Reply:
@nordic-krch what do you think of this proposal?

Reply:
Sorry to interrupt. The current Gazell commit sequence is not optimal.

The original line is as follows:
`depends on HAS_HW_NRF_TIMER0`
The first commit changes the line to:
`depends on !$(dt_nodelabel_enabled,timer0)`
The second commit changes the line to:
`depends on $(dt_nodelabel_has_compat,timer0,$(DT_COMPAT_NORDIC_NRF_TIMER))`

The second commit serves the same purpose as the original line. It checks if the nRF52 SoC has the corresponding TIMER instance. This patch should be the first commit rather than the second one.

The first commit adds some protection to the timer resource. The original line did not check the Zephyr timer driver state. I was unaware of the Kconfig NRF_HW_TIMER0_RESERVED option.

IMO, the first commit should be replaced by the second one. It is suggested the first commit is added afterwards.

Reply:
@nordic-krch ping

Reply:
I am happy to do as you suggest @leewkb4567 -- would just like @nordic-krch to confirm it'll work for him as well; I know he's using something similar to these commits elsewhere so I want to make sure we're all three on the same page